ovsdb/client.go

```diff
@@ -133,29 +133,23 @@ func (c *Client) rpc(method string, out interface{}, args ...interface{}) error
 	}
 
 	// Add callback for this RPC ID to return results via channel.
-	ch := make(chan rpcResponse, 0)
+	ch := make(chan rpcResponse, 1)
+	defer close(ch)
 	c.addCallback(req.ID, ch)
 
 	if err := c.c.Send(req); err != nil {
 		return err
 	}
 
-	// Wait for callback to fire.
-	res := <-ch
-	if err := res.Error; err != nil {
-		return err
+	// Await RPC completion or cancelation.
+	select {
+	case <-ctx.Done():
+		// RPC canceled; clean up callback.
+		return c.cancelCallback(ctx, req.ID)
+	case res := <-ch:
```

Comment:
Do you want to take care of the channel close here?

Reply:
I actually deferred closing the channel on line 137!

Reply:
Oh, I mean *closed channel*, as `case res, ok := <-ch:`.  We can take care of it separately.

Reply:
Should be no need as far as I can tell.  The channel is only ever closed by this function, unless I'm missing something.

Reply:
Oh, I see.  The thing is closing the channel by the consumer side is a bit risky, as it will cause a crash when the producer send a message on a closed channel.  In a fully concurrent environment, there is no gurantee to avoid that situation and letting producer close the channel is one way to avoid that.

It would be cool to take that into account in the future PR.